python/cudf/cudf/core/column/numerical.py

```diff
@@ -471,14 +471,17 @@ def fillna(
         if method is not None:
             return super(NumericalColumn, col).fillna(fill_value, method)
 
+        if fill_value is None:
+            raise ValueError("Must specify either 'fill_value' or 'method'")
```

Comment:
Do we actually not allow `None` here or does that mean fill with `null` values?

Reply:
`None` is allowed when you use `method`:

```
In [8]: s = cudf.Series([1, 2, None, None, 3])

In [9]: s.fillna(method='ffill')
Out[9]:
0    1
1    2
2    2
3    2
4    3
dtype: int64
```

Reply:
Sorry - I misunderstood the question. We should explicitly _not_ allow `None` when `method=None`. 

If someone wants to fillna() with null values, they should just do nothing instead :)

Reply:
> If someone wants to fillna() with null values, they should just do nothing instead :)

This isn't true in the case of `float` dtype columns though, no? Since we handle both `NaN` and `null` values via `fillna` in that case.

Reply:
What case are we thinking, where someone wants to convert `nans` to nulls? Or nulls to nans?

Reply:
In the former case, they could use [`nans_to_nulls()`](https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/core/series.py#L1960).

In the latter case, they could use `fillna(np.nan)` which is much more explicit.

FWIW, we currently throw when both `method=None` and `fill_value=None` further down: https://github.com/rapidsai/cudf/blob/branch-0.18/python/cudf/cudf/_lib/replace.pyx#L134